mysite/lti/views.py

```diff
@@ -0,0 +1,115 @@
+import pickle
+import oauth2
+import json
+from django.template import RequestContext
+from django.core.urlresolvers import reverse
+from django.views.decorators.csrf import csrf_exempt
+from ims_lti_py.tool_provider import DjangoToolProvider
+from django.shortcuts import render_to_response, redirect
+
+from lti.models import LTIUser
+from lti import app_settings as settings
+
+
+ROLES = {
+    'Instructor': 'prof',
+    'Leaner': 'student',
+}
+
+MOODLE_PARAMS = (
+    'user_id',
+    'ext_lms',
+    'lis_person_name_full',
+    'lis_person_name_given',
+    'lis_person_name_family',
+    'lis_person_contact_email_primary',
+)
+
+
+@csrf_exempt
+def lti_init(request, course_id=None, unit_id=None):
+    """LTI init view
+
+    |  Analyze LTI POST request to start LTI session.
+    |  Create LTIUser with all needed link to Django user and/or UserSocialAuth.
+    |  Finally login Django user.
+
+    :param course_id: course id from launch url
+    :param unit_id: unit id from lunch url
+    """
+    if settings.LTI_DEBUG:
+        print('META')
+        print request.META
+        print('PARAMS')
+        print request.POST
+    session = request.session
+    session.clear()
+    try:
+        consumer_key = settings.CONSUMER_KEY
+        secret = settings.LTI_SECRET
+
+        tool = DjangoToolProvider(consumer_key, secret, request.POST)
+        is_valid = tool.is_valid_request(request)
+        session['message'] = "We are cool!"
+        session['target'] = '_blank'
+    except (oauth2.MissingSignature,
+            oauth2.Error,
+            KeyError,
+            AttributeError) as e:
+        is_valid = False
+        session['message'] = "{}".format(e)
+        pass
+    session['is_valid'] = is_valid
+    # copy request to dictionary
+    request_dict = dict()
+    for r in request.POST.keys():
+        request_dict[r] = request.POST[r]
+    session['LTI_POST'] = pickle.dumps(request_dict)
+
+    '------------------------------------------------------------------------'
+    consumer_name = request_dict.get('ext_lms', 'lti')
+    user_id = request_dict.get('user_id', None)
+    roles = request_dict.get('roles', None)
```

Comment:
> Better change roles to roles_from_request or smth like that. It is bad practice to use same variable for different values, because it is hard to debug, or see from possible error message, at which step roles is wrong.

Need to discuss.